### PR TITLE
SDA2020: TASK 2 - oop part 1

### PR DESCRIPTION
**Pull request by BohdanShuliaka**

### CODE REVIEW

File: 02_oop_part_1/code/src/Inventory.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import { ItemComparator } from './ItemComparator';
+
+export class Inventory {
+    items: Item[] = [];
+
+    addItem(item: Item) {
+        this.items.push(item)
+    }
+
+    sort(comparator?: ItemComparator): void {
```
Comment (dianalemen) [inline 02_oop_part_1/code/src/Inventory.ts:13]: I believe, in this case, it has to be function overload

Inventory.sort(): This sorts the items in the Inventory instance based on their value
Inventory.sort(comparator: ItemComparator): This sorts the items in the Inventory instance based on their weight.

File: 02_oop_part_1/code/src/Sword.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Weapon } from './Weapon';
+
+export class Sword extends Weapon {
+
+    constructor(baseDamage: number, baseDurability: number, value: number, weight: number) {
+        super('sword', baseDamage, baseDurability, value, weight)
+    }
+
+    polish() {
+        if(this.damageModifier < (this.baseDamage * 0.25)) {
```
Comment (dianalemen) [inline 02_oop_part_1/code/src/Sword.ts:10]: Where do you have this implementation -  If the base damage of a sword were to be 100, then the maximum that the effective damage could be increased to would be 125. ?

**Comment by dianalemen** [comment]
Also, please, check the UML carefully. All of the methods have the return type. Fix it in all classes that need it.
image

File: 02_oop_part_1/code/src/Inventory.ts (hunk 1 above)
Comment (BohdanShuliaka) [reply 02_oop_part_1/code/src/Inventory.ts:13]: yeah but as we agreed on previous Q&A that it is similar way of implementation, so I just used optional param. Is is critical?

File: 02_oop_part_1/code/src/Sword.ts (hunk 2 above)
Comment (BohdanShuliaka) [reply 02_oop_part_1/code/src/Sword.ts:10]: this is the implementation, our damageModifier  cannot be more than 25% of base damage, and increasing is made on the next line... dont see the reason to make some new methods

File: 02_oop_part_1/code/src/Inventory.ts (hunk 1 above)
Comment (BohdanShuliaka) [reply 02_oop_part_1/code/src/Inventory.ts:13]: done :)

Comment (dianalemen) [reply 02_oop_part_1/code/src/Inventory.ts:13]: it works the same, but there is an requirement in the task ;)

File: 02_oop_part_1/code/src/Sword.ts (hunk 2 above)
Comment (dianalemen) [reply 02_oop_part_1/code/src/Sword.ts:10]: the question wasn’t about new methods. It was about the implementation itself, but actually, I got it the wrong way, so this implementation is enough.

**Review by dianalemen — APPROVED** [review]

**Pull request merged**